app/models/conversion_host.rb

```diff
@@ -151,13 +151,21 @@ def apply_task_limits(path, limits = {})
     raise "Could not apply the limits in '#{path}' on '#{resource.name}' with [#{err.class}: #{err}]"
   end
 
+  # Run the virt-v2v-wrapper.py script on the remote host and return a hash
+  # result from the parsed JSON output.
+  #
+  # Certain sensitive fields are filtered in the error messages to prevent
+  # that information from showing up in the UI or logs.
+  #
   def run_conversion(conversion_options)
+    ignore = %w[password fingerprint key]
+    filtered_options = conversion_options.clone.tap { |h| h.each { |k, _v| h[k] = "__FILTERED__" if ignore.any? { |i| k.to_s.end_with?(i) } } }
```

Comment:
What do you think of this?
```ruby
filtered_options = conversion_options.each_with_object({}) { |(k, v), h| h[k] = (k.to_s =~ /(fingerprint|key|password)\z/) ? "__FILTERED__" : v }
```

Reply:
Looks more difficult than mine. I'm sure there are multiple ways to skin a cat here, but I'd like to keep mine unless you have strong objections.